camayoc/tests/qpc/api/v1/conftest.py

```diff
@@ -17,51 +17,32 @@
     Source,
 )
 from camayoc.tests.qpc.api.v1.utils import wait_until_state
-from camayoc.tests.utils import get_vcenter_vms, wait_until_live
+from camayoc.tests.utils import (
+    get_vcenter_vms,
+    power_off_vms,
+    power_on_vms,
+)
 from camayoc.utils import run_scans
 
 
 SCAN_DATA = {}
 """Cache to associate the named scans with their results."""
 
 
-def create_cred(cred_info, session_cleanup):
+def create_cred(cred_info, cleanup):
     """Given info about cred from config file, create it and return id."""
-    c = Credential(
-        cred_type=cred_info['type'],
-        username=cred_info['username'],
-    )
-    if cred_info.get('sshkeyfile'):
-        c.ssh_keyfile = cred_info['sshkeyfile']
-    else:
-        c.password = cred_info['password']
+    c = Credential(**cred_info)
     c.create()
-    session_cleanup.append(c)
+    cleanup.append(c)
     return c._id
 
 
-def create_source(source_info, cred_name_to_id_dict, session_cleanup):
-    """Given info about source from config file, create it.
-
-    :returns: A tuple containing the The id of the created source and a
-        dictionary of the expected products from the config file.
-    """
-    cred_ids = [value for key, value in cred_name_to_id_dict.items(
-    ) if key in source_info['credentials']]
-    host = source_info.get('ipv4') if source_info.get(
-        'ipv4') else source_info.get('hostname')
-    src = Source(
-        source_type=source_info.get('type', 'network'),
-        hosts=[host],
-        credential_ids=cred_ids,
-        options=source_info.get('options')
-    )
-    expected_products = source_info.get('products', {})
-    expected_products.update(
-        {'distribution': source_info.get('distribution', {})})
+def create_source(source_info, cleanup):
+    """Given info about source from config file, create it and return id."""
+    src = Source(**source_info)
     src.create()
-    session_cleanup.append(src)
-    return src._id, expected_products
+    cleanup.append(src)
```

Comment:
If I remember correctly, in the past we were concerned about cleaning up creds causing something with the reports getting cleaned up too, is that resolved? If it was, woo!

Reply:
I will run the entire test suite to ensure I am not breaking anything. Will be watching for this.

Reply:
Ok cool.

Reply:
yeah was concerned about deleting credentials cascading into deleting sources and scans and thus reports
but maybe reports can be orphaned

Reply:
Going to approve PR, trusting that you will determine if this is now safe!

Reply:
I've checked and removing the entities does not impact on the report tests.